### PR TITLE
[Chore][Bench] Add benchmark for MeanPoolingForwardOp

### LINKED ISSUE #261: [Benchmark] Add benchmark for MeanPoolingForwardOp
Parent: #247

## Description

`MeanPoolingForwardOp` is the only op that has **no benchmark file at all**. A new benchmark file needs to be created from scratch.

## TODO

- Create `benchmarks/ops/bench_mean_pooling.py`
- Define `MeanPoolingBenchmark` class with `calculate_flops()` and `calculate_memory()` methods
- Add `test_mean_pooling_bench()` function
- Profile TileOPs `MeanPoolingForwardOp` via `bm.profile(op, *inputs)`
- Profile a baseline implementation (e.g., PyTorch mean pooling)
- Ensure FLOPS and memory metrics are computed correctly

## Acceptance Criteria

The PR must include `profile_run.log` output for both **baseline** and **tileops**, demonstrating that the benchmark actually runs and produces reasonable results.

### PR DESCRIPTION
**Pull request by zhen8838**
## Summary

- Adds `benchmarks/ops/bench_mean_pooling.py` with `MeanPoolingBenchmark` class implementing `calculate_flops()` and `calculate_memory()`
- Profiles both TileOPs `MeanPoolingForwardOp` and PyTorch baseline across 6 test cases covering uniform and variable-length (offset-based) chunking scenarios
- All 6 test cases pass

Closes #261

### tileops

| batch_size | seq_len | heads | dim | chunk_size | dtype | accum_dtype | chunks_per_bacth | seq_num | use_offsets | latency_ms | tflops | bandwidth_gbs |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 1 | 8192 | 64 | 128 | 64 | torch.float16 | torch.float32 | 128 | 1 | 0 | 1.20 | 0.06 | 0.11 |
| 1 | 8192 | 64 | 128 | 64 | torch.float16 | torch.float32 | 128 | 1 | 0 | 0.87 | 0.08 | 0.16 |
| 2 | 2048 | 64 | 128 | 64 | torch.float16 | torch.float32 | 32 | 2 | 0 | 0.61 | 0.05 | 0.11 |
| 1 | 1024 | 64 | 128 | 64 | torch.float16 | torch.float32 | 16 | 3 | 1 | 0.17 | 0.05 | 0.10 |
| 1 | 8192 | 64 | 128 | 64 | torch.float16 | torch.float32 | 128 | 4 | 1 | 1.07 | 0.06 | 0.13 |
| 1 | 1000 | 64 | 128 | 32 | torch.float16 | torch.float32 | 34 | 4 | 1 | 0.10 | 0.09 | 0.16 |

### baseline

| batch_size | seq_len | heads | dim | chunk_size | dtype | accum_dtype | chunks_per_bacth | seq_num | use_offsets | latency_ms | tflops | bandwidth_gbs |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 1 | 8192 | 64 | 128 | 64 | torch.float16 | torch.float32 | 128 | 1 | 0 | 0.78 | 0.09 | 0.17 |
| 1 | 8192 | 64 | 128 | 64 | torch.float16 | torch.float32 | 128 | 1 | 0 | 0.78 | 0.09 | 0.17 |
| 2 | 2048 | 64 | 128 | 64 | torch.float16 | torch.float32 | 32 | 2 | 0 | 0.33 | 0.10 | 0.21 |
| 1 | 1024 | 64 | 128 | 64 | torch.float16 | torch.float32 | 16 | 3 | 1 | 0.12 | 0.07 | 0.15 |
| 1 | 8192 | 64 | 128 | 64 | torch.float16 | torch.float32 | 128 | 4 | 1 | 0.78 | 0.09 | 0.18 |
| 1 | 1000 | 64 | 128 | 32 | torch.float16 | torch.float32 | 34 | 4 | 1 | 0.24 | 0.04 | 0.07 |

## Checklist

- [x] Pre-commit passes
- [x] All 6 benchmark test cases pass
- [x] Only `benchmarks/ops/bench_mean_pooling.py` is added (no generated files committed)
- [x] Profile run output included above

### CODE REVIEW

File: benchmarks/ops/bench_mean_pooling.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        use_offsets = 1
+    else:
+        offsets = torch.arange(
+            0, (batch_size + 1) * seq_len,
+            seq_len,
+            dtype=torch.int32,
+            device='cuda',
+            requires_grad=False)
+        chunks_per_bacth = (seq_len + chunk_size - 1) // chunk_size
+        indices = torch.randint(0, seq_len, (chunks_per_bacth, 2), dtype=torch.int32, device='cuda')
```
[the remark this answers is not in the document]
Comment (zhen8838) [reply benchmarks/ops/bench_mean_pooling.py]: Accepted. Changed `torch.randint` to `torch.empty` since the `indices` tensor is unused when `use_offsets=0`. See ca360b7.

File: benchmarks/ops/bench_mean_pooling.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    bm = MeanPoolingBenchmark(test)
+    inputs = test.gen_inputs()
+
+    op = MeanPoolingForwardOp(**params)
+    result = bm.profile(op, *inputs)
+    BenchmarkReport.record("mean_pooling", locals(), result, tag="tileops")
+
+    result_bl = bm.profile(test.ref_program, *inputs)
+    BenchmarkReport.record("mean_pooling", locals(), result_bl, tag="baseline")
```
[the remark this answers is not in the document]
Comment (zhen8838) [reply benchmarks/ops/bench_mean_pooling.py:79]: Declined. Using `locals()` is the established convention in existing benchmarks (see `bench_gemm.py:30`, `bench_gqa.py`). `BenchmarkReport.record` already filters params via `filtered_params` in `benchmark.py:75-80`, so passing `locals()` is safe and consistent with the codebase pattern.

**Review by superAngGao — APPROVED** [review]
LGTM.
